gapic/src/main/com/google/gapid/models/ApiContext.java

```diff
@@ -204,6 +211,20 @@ public boolean contains(long index) {
       return Ranges.contains(context.getRangesList(), index) >= 0;
     }
 
+    public CommandRange findClosest(CommandRange range) {
+      int index = Ranges.contains(context.getRangesList(), last(range));
+      if (index >= 0) {
+        return commands(last(range), 1);
+      } else if (index == -1) {
+        // The given range ends before any of our commands.
+        return (context.getRangesCount() == 0) ? null : commands(first(context.getRanges(0)), 1);
+      } else {
```

Comment:
Yes. It's the most commonly executed case. 
Index is positive if the context happens to contain the given range.
Index is -1 if the given range is before the context is created.
Index is < -1 if the given range is after the context is created, but the context does not contain the range.